wrapcheck/wrapcheck.go

```diff
@@ -98,24 +98,27 @@ func NewAnalyzer(cfg WrapcheckConfig) *analysis.Analyzer {
 }
 
 func run(cfg WrapcheckConfig) func(*analysis.Pass) (interface{}, error) {
-	// Precompile the regexps, report the error
-	var (
-		ignoreSigRegexp        []*regexp.Regexp
-		ignoreInterfaceRegexps []*regexp.Regexp
-		ignorePackageGlobs     []glob.Glob
-		err                    error
-	)
-
-	ignoreSigRegexp, err = compileRegexps(cfg.IgnoreSigRegexps)
-	if err == nil {
-		ignoreInterfaceRegexps, err = compileRegexps(cfg.IgnoreInterfaceRegexps)
-	}
-	if err == nil {
-		ignorePackageGlobs, err = compileGlobs(cfg.IgnorePackageGlobs)
-
-	}
 
 	return func(pass *analysis.Pass) (interface{}, error) {
+		// Precompile the regexps, report the error
```

Comment:
The downside of having the compilation steps here is that this anonymous function will be executed once per package. Having it within the body of `run` executes it once on startup. Now, this could completely be a useless optimisation, but I don't have any numbers on hand to make a strong argument either way. Hence, doing the compilation only once steers on the safer side.

Do you have any reasoning for shifting it down here?

Reply:
hello, @tomarrell !

I'm very excited you can reply me.

I agree with you, this anonymous function will be executed in per package. My original intention is fix the error check.

I had an illusion after seeing testdata dir. i thought every package has a `.wrapcheck.yaml`, in general, one project only have one `.wrapcheck.yaml`.

so, what do you think error check in the follow:

```Go
func run(cfg WrapcheckConfig) func(*analysis.Pass) (interface{}, error) {
	// Precompile the regexps, report the error
	var (
		ignoreSigRegexp        []*regexp.Regexp
		ignoreInterfaceRegexps []*regexp.Regexp
		ignorePackageGlobs     []glob.Glob
		err                    error
	)

	ignoreSigRegexp, err = compileRegexps(cfg.IgnoreSigRegexps)
	if err != nil {
		log.Fatalf("compile IgnoreSigRegexps failed: %v", err)
	}

	ignoreInterfaceRegexps, err = compileRegexps(cfg.IgnoreInterfaceRegexps)
	if err != nil {
		log.Fatalf("compile IgnoreInterfaceRegexps failed: %v", err)
	}
        // ...
}
```

Reply:
Apologies for the late reply, I've had a work event and some holiday the past week.

The adjustment looks good to me.

Reply:
Hello @tomarrell Thanks for the reply. 

I have already revert the change of reg compile code location.I can't find a new way that is more better to fix the error check. 

Thanks.